packages/lib/BaseModel.ts

```diff
@@ -754,6 +756,13 @@ class BaseModel {
 		return this.db_;
 	}
 
+	public static generateUuid() {
+		return this.uuidGenerator();
+	}
+
+	public static setIdGenerator(generator: ()=> string) {
+		this.uuidGenerator = generator;
```

Comment:
A good pattern for this kind of setter is to return the previous generator so that it can be reset by the caller. That's more future proof.